lib/ssl/src/ssl_certificate.erl

```diff
@@ -61,20 +61,22 @@
 %% Note: Path = lists:reverse(Chain) -- Root, that is on the peer cert 
 %% always comes first in the chain but last in the path.
 %%--------------------------------------------------------------------
-trusted_cert_and_paths([Peer] = Chain,  CertDbHandle, CertDbRef, PartialChainHandler) ->
+trusted_cert_and_paths([Peer],  CertDbHandle, CertDbRef, PartialChainHandler) ->
     OtpCert = public_key:pkix_decode_cert(Peer, otp),
+    Chain = [{Peer, OtpCert}],
```

Comment:
What about using a record here like -record(cert, {der, decoded}) wouldn't that be easier to match and easier to detect errors in compile time with? Never mind the names, they could perhaps be better.

Reply:
I thought about that and I was lazy, you get so long lines when you pick them apart, but maybe we should change
that.

Reply:
I think it can be hard to remember which part of the tuple is the decoded part and which one is the other and it could be mixed up when matching other tuples in function heads.

Reply:
The principal have been to call the decode cert OtpCert as in reference to the ASN-1 type of the decode certs. Although I see that it has in some places been called ElrCert instead so it could be a good idea to clarify this in an other way.